src/gvars.c

```diff
@@ -226,6 +226,67 @@ static Obj             FopiesGVars;
 
 #endif
 
+// FlagsGVars contains information about global variables as a single INTOBJ
+// Once turned into a C int, this information is:
+//
+// Bits 0 and 1 - A value of type GVarWriteFlag which denotes if the variable
+// is ReadOnly, Mutable or Constant
+// Bit 2 - If the variable has ever had a non-default value assigned to
+// ExprGVars,
+//         CopiesGVars or FopiesGVars. Note that this bit is never cleared at
+//         present, so it can be set to 1 for a gvar, while these three arrays
+//         all have their default value.
+
+// Below are some helper methods for reading and writing FlagsGVars. They
+// should always be used to simplifying adding more flags in future.
+
+typedef enum {
+    GVarReadOnly = 0,
+    GVarMutable = 1,
+    GVarConstant = 2
+} GVarWriteFlag;
+
+static GVarWriteFlag GetGVarWriteState(Int gvar)
+{
+    UInt val = INT_INTOBJ(ELM_GVAR_LIST(FlagsGVars, gvar));
+    return val & 3;
+}
+
+typedef struct {
+    GVarWriteFlag gvarWriteFlag;
+    Int           hasExprCopiesFopies;
+} GVarFlagInfo;
```

Comment:
So this struct takes 16 bytes on a 64 bit system to store 3 bits of information, which strikes me as a tad wasteful :-). How about this instead:
```C
typedef struct {
    unsigned gvarWriteFlag : 2;
    unsigned hasExprCopiesFopies : 1;
} GVarFlagInfo;
```
And then one could simplify `GetGVarWriteState`, `GetGVarFlagInfo`, etc. ... Well, maybe it wouldn't really be simpler? I must admit I haven't quite thought this through... will do it tomorrow, after some sleep :-)

Reply:
I wondered if I could just cast a struct last this to/from an int, but I suspect by the C standard that isn't legal.

Reply:
That's what we have `union` for, though ;-). So either
```C
  union {  Int intval;  GVarFlagInfo flags; };
```
or directly inside `GVarFlagInfo`:
```C
typedef union {
    int intval;
    struct {
        unsigned gvarWriteFlag : 2;
        unsigned hasExprCopiesFopies : 1;
    };
} GVarFlagInfo;
```

Reply:
Alternatively, on can of course also cheat and do  `Int val = ...;  return *(GVarFlagInfo *)&val;` but that seems more fragile.

Reply:
So in the end I went with memcpy, while I admit it looks stupid, in my experience it seems like the most reliable long-term solution to doing this kind of type changing.